### PR TITLE
KT-20357: Add samples for joinToString, joinTo

### PR DESCRIPTION
**Pull request by gzoritchak**

### CODE REVIEW

File: libraries/stdlib/samples/test/samples/collections/collections.kt [hunk 1]
```diff
@@ -311,6 +311,28 @@ class Collections {
             assertPrints(moreFrequencies, "{o=1, t=4, f=2, s=2, e=2, n=1}")
         }
 
+        @Sample
+        fun joinTo() {
+            val sb = StringBuilder("An existing string and a list:")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/collections.kt]: minor: add a space after colon

File: libraries/stdlib/samples/test/samples/collections/collections.kt [hunk 2]
```diff
@@ -311,6 +311,28 @@ class Collections {
...
+            val numbers = listOf(1, 2, 3)
+            assertPrints(numbers.joinTo(sb, prefix = "[", postfix = "]").toString(), "An existing string and a list:[1, 2, 3]")
+            
+            val lotOfNumbers = (1..100).asIterable()
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/collections.kt]: `IntRange` is already an `Iterable`. What does this conversion intend to show?

Comment (gzoritchak) [reply libraries/stdlib/samples/test/samples/collections/collections.kt]: Yes, it's maybe more clean that way:
```
val lotOfNumbers: Iterable<Int> = 1..100
```

**Comment by ilya-g** [comment]
Thanks, I've merged it.

**Pull request merged**